pkg/kube/client.go

```diff
@@ -695,7 +696,22 @@ func (c *client) EnvoyDo(ctx context.Context, podName, podNamespace, method, pat
 
 func (c *client) portForwardRequest(ctx context.Context, podName, podNamespace, method, path string, port int) ([]byte, error) {
 	formatError := func(err error) error {
-		return fmt.Errorf("failure running port forward process: %v", err)
+		pods, _ := c.GetIstioPods(context.TODO(), podNamespace, map[string]string{
```

Comment:
Can we just parse the error message or something? Doing a lookup on pods just to return a better error message seems like a high cost

Reply:
sure, that's okay

Reply:
Hi @howardjohn,  I have checked that
Firstly,  there may be different output for 3 cases:
1. Docker as a container runtime
2. containerd as a container runtime but version is less than 1.50
3. containerd as a container runtime but version is equal or great than 1.50
It's easy to identify case 1 which use **socat**, but parse the error message may be not enough to distinguish 2 and 3 which just tell user the connection refuse. 
Secondly, this lookup on pods could be happened only when error is occurring, not every time.
Thirdly, it may be not reliable to depend on the error output from other, such as container runtime or kubelet.
So can we keep the current logic for this?

Reply:
I feel very uncomfortable running remote calls in error handling.. Its quite unexpected that when I do a port forward call suddenly we are calling LIST PODS (very expensive, may not have permissions). IMO, I would rather make no change than do this lookup.

Reply:
Hi @howardjohn,  how about just print out the notification when there is error? There would be no more other cost and risk. does it make sense?

Reply:
sgtm

Reply:
okay